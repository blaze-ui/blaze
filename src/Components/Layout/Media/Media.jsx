import React,{Component} from 'react';
import Proptypes from "prop-types";

export default class Media extends Component{
    render() {
        if(this.props.typeGet === "linear"){
                   return( <div style={{display:'flex'}}>
                    <div>
                        <img height={64} width={64} src="https://tinyurl.com/yylsuh97" alt="Generic placeholder"/>
                    </div>
                    <div>
                       <h3>Media Heading</h3>
                       <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book</p>
                    </div>                   
                   </div>   )
        }else if(this.props.typeGet === "nested"){
            return(
            <div>
                <img height={64} width={64} src="https://tinyurl.com/yylsuh97" alt="Generic placeholder"/>
                <div>
                    <h3>Hello</h3>
                    <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book</p>
    
                    <div style={{paddingLeft:'10vw'}}>
                        <img height={64} width={64} src="https://tinyurl.com/yylsuh97" alt="Generic placeholder"/>
                         <div>
                            <h3>Nested Hola</h3>
                            <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book</p>
                        </div>    
                    </div>
                 </div>                
            </div>
            );
        }else{
            console.log('Error');
        }
           
        
    }
}
Media.propType = {
    typeGet: Proptypes.oneOf(['linear','nested'])
  }