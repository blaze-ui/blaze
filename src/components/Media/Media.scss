@import "../../styles/ColorSchema.scss";

.Media {
    display: flex; 
    color: #000; 
}

.Media p {
  line-height: 22px;
}

.Media h1 {
    margin: 1em 0 0.5em 0;
	color: #4d0000;
	font-weight: normal;
	font-family: 'Ultra', sans-serif;   
	font-size: 36px;
	line-height: 42px;
    text-transform: uppercase;
    text-decoration-line: underline;
	text-shadow: 1px 2px white, 2px 2px yellowgreen;
}

.Media img {
    height: 10rem;
    width: 10rem;
    margin: 0 1rem;
    transition: .3s ease-in-out;
}

.Media img:hover {
	transform: scale(1.3);
}


  .mediaNested, .mediaLinear {
    margin: 3rem 7rem;
    padding: 2rem;
    border: 1px solid teal;
    background: #fff;
    padding: 30px;
    position: relative;
  }
  
  .mediaNested, .mediaLinear,
  .mediaLinear::before,
  .mediaLinear::after,
  .mediaNested::before,
  .mediaNested::after {
    /* Styles to distinguish sheets from one another */
    box-shadow: 1px 1px 1px rgba(0,0,0,0.25);
    border: 1px solid #bbb;
  }
  
  .mediaLinear::before,
  .mediaLinear::after,
  .mediaNested::before,
  .mediaNested::after {
    content: "";
    position: absolute;
    height: 95%;
    width: 99%;
    background-color: #eee;
  }
  
  
  .mediaLinear::before,
  .mediaNested::before {
    right: 15px;
    top: 0;
    transform: rotate(-1deg);
    z-index: -1;
  }
  
  .mediaLinear::after,
  .mediaNested::after {
    top: 5px;
    right: -5px;
    transform: rotate(1deg);
    z-index: -2;
  }