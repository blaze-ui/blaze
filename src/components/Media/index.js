/* eslint-disable react/no-access-state-in-setstate */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from "react";
import Proptypes from "prop-types";
import "./Media.css";
function Media(props) {
  const { typeGet, image } = props;
  const content = props.children;
  return (
    <div>
      {/* {
              (typeGet === "linear") ? 
              <Linear value={typeGet} content={content}/> :
              <Nested value={typeGet} content={content}/>
            } */}
      <Rendering typeGet={typeGet} content={content} image={image} />
    </div>
  );
}

function Rendering({ typeGet, content, image }) {
  if (typeGet === "linear") {
    return <Linear value={typeGet} content={content} image={image} />;
  } else if (typeGet === "nested") {
    return <Nested value={typeGet} content={content} image={image} />;
  } else {
    console.log("props error");
  }
}

const Linear = ({ props, content, image }) => {
  return (
    <div className="mediaLinear">
      <div className="Media">
        <div>
          <img src={image} alt="Generic placeholder" />
        </div>
        <div>
          <h1>Media Heading</h1>
          <p>{content}</p>
        </div>
      </div>
    </div>
  );
};

const Nested = ({ props, content, image }) => {
  return (
    <div className="mediaNested">
      <div className="Media">
        <div>
          <img src={image} alt="Generic placeholder" />
        </div>
        <div>
          <h1>Heading 1</h1>
          <p>
            There are many variations of passages of Lorem Ipsum available, but
            the majority have suffered alteration in some form, by injected
            humour, or randomised words which don't look even slightly
            believable. If you are going to use a passage of Lorem Ipsum, you
            need to be sure there isn't anything embarrassing hidden in the
            middle of text. All the Lorem
          </p>
        </div>
      </div>
      <div style={{ paddingLeft: "10vw" }}>
        <div className="Media">
          <div>
            <img src={image} alt="Generic placeholder" />
          </div>
          <div>
            <h1>Heading 2</h1>
            <p>
              Contrary to popular belief, Lorem Ipsum is not simply random text.
              It has roots in a piece of classical Latin literature from 45 BC,
              makin g it over 2000 years old. Richard McClintock, a Latin
              professor at Hampden-Sydney College in Virginia, looked up one of
              the more obscure Latin words, consectetur, from a Lorem Ipsum
              passage, and going through the cites of the word in classical
              literature,
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};

Media.propType = {
  typeGet: Proptypes.oneOf(["linear", "nested"]),
  image: Proptypes.string
};

export default Media;
