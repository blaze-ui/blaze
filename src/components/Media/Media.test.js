import React from "react";
import renderer from "react-test-renderer";
import Media from "./index";

test("Media default test case", () => {
  const component = renderer.create(
    <Media typeGet="linear">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      Lorem Ipsum has been the industry's standard dummy.
    </Media>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
