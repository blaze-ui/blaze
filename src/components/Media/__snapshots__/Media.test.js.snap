// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media default test case 1`] = `
<div>
  <div
    className="mediaLinear"
  >
    <div
      className="Media"
    >
      <div>
        <img
          alt="Generic placeholder"
        />
      </div>
      <div>
        <h1>
          Media Heading
        </h1>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy.
        </p>
      </div>
    </div>
  </div>
</div>
`;
